openwisp_users/models.py

```diff
@@ -1,133 +1,41 @@
-    def _create_user(self, *args, **kwargs):
-        """
-        adds automatic email address object creation to django
-        management commands "create_user" and "create_superuser"
-        """
-        user = super()._create_user(*args, **kwargs)
-        self._create_email(user)
-        return user
-
-    def _create_email(self, user):
-        """
-        creates verified and primary email address objects
-        """
-        if user.email:
-            set_primary = (
-                EmailAddress.objects.filter(user=user, primary=True).count() == 0
-            )
-            email = EmailAddress.objects.create(
-                user=user, email=user.email, verified=True
-            )
-            if set_primary:
-                email.set_as_primary()
+from .base.models import (
+    AbstractUser,
+    BaseGroup,
+    BaseOrganization,
+    BaseOrganizationOwner,
+    BaseOrganizationUser,
+)
 
 
 class User(AbstractUser):
-    """
-    OpenWISP User model
-    """
-
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    email = models.EmailField(_('email address'), unique=True, blank=True, null=True)
-    bio = models.TextField(_('bio'), blank=True)
-    url = models.URLField(_('URL'), blank=True)
-    company = models.CharField(_('company'), max_length=30, blank=True)
-    location = models.CharField(_('location'), max_length=128, blank=True)
-    phone_number = PhoneNumberField(unique=True, blank=True, null=True)
-
-    objects = UserManager()
-
     class Meta(AbstractUser.Meta):
-        index_together = ('id', 'email')
-
-    @cached_property
-    def organizations_pk(self):
-        """
-        returns primary keys of organizations the user is associated to
-        """
-        manager = OrganizationUser.objects
-        qs = (
-            manager.filter(user=self, organization__is_active=True)
-            .select_related()
-            .only('organization_id')
-            .values_list('organization_id')
-        )
-        return qs
-
-    def clean(self):
-        if self.email == '':
-            self.email = None
-
-
-class Group(BaseGroup):
-    """
-    Proxy model used to move ``GroupAdmin``
-    under the same app label as the other models
-    """
-
-    class Meta:
-        proxy = True
-        verbose_name = _('group')
-        verbose_name_plural = _('groups')
-
-
-class Organization(AbstractOrganization):
-    """
-    OpenWISP Organization model
-    """
-
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    description = models.TextField(_('description'), blank=True)
-    email = models.EmailField(_('email'), blank=True)
-    url = models.URLField(_('URL'), blank=True)
+        abstract = False
 
-    def __str__(self):
-        value = self.name
-        if not self.is_active:
-            value = '{0} ({1})'.format(value, _('disabled'))
-        return value
 
+class Organization(BaseOrganization, AbstractOrganization):
```

Comment:
I had to use `class Organization(BaseOrganization, AbstractOrganization)` here because if I try to do something like:

```
from organizations.abstract import AbstractOrganization

class BaseOrganization(AbstractOrganization):
   # Module changes    
   class Meta:
         abstract = True

class Organization(BaseOrganization):
   class Meta:
         abstract = False
```

I get error stating `BaseOrganization` should not be abstract. 
I tried to find the reason in the docs of `django-organization` and I *think* if we want to make changes in Organzation's Abstract class, we need to use `[OrganizationBase](https://django-organizations.readthedocs.io/en/latest/cookbook.html?highlight=Abstract#difference-between-abstract-and-base-models)` which requires us to copy a lot of code from django-organization.
I tried to look into django-organization's code as well, but I can't find the line that's causing it to think `BaseOrganization` is the concrete class, hence this solution.

If you have any information about this and can give me pointers of where I should look to find the cause of the error, that will be very helpful.